### ISSUE TITLE
Gate.control() method rasies the error 'Cannot make controlled gate with id instruction'

### ISSUE THREAD

**Issue by ljubabu** [Qiskit/qiskit#4010]
### Information

- **Qiskit Terra version**:
0.12.0
- **Python version**:
Python 3.7.3
- **Operating system**:
WIN10

### What is the current behavior?
Converting circuit into gate using the `QuantumCircuit.to_gate` property, and then applying the `Gate.control` method raises the error 'Cannot make controlled gate with id instruction'. This actually means that you cannot create a control gate if you have an id instruction in your circuit. However, when the circuit with no `IdGate` is used, the Gate.control work just fine.

### Steps to reproduce the problem

```
CIRCF_2.iso(B2, [0, 1, 2, 3],[]) # works fine 
GATEB2=transpile(CIRCF_2, basis_gates=['u1', 'u2', 'u3', 'cx', 'id']).to_gate() # works fine
CGA=GATEB2.control(1) # crashes with error 'Cannot make controlled gate with id instruction'
```

### What is the expected behavior?
The Identity gate `IdGate` should not compromise the control instruction.

**Comment by ewinston** [comment]
I believe this should be resolved by pr #3950.

**Comment by mrvee-qC** [comment]
# Update for Santa:elf:- QGoGP
:no_entry_sign: I am not sure what isometry "B2" refers to but I tried out an arbitary isometry and it seems that the issue is fixed. Have attached the code I used and screenshots for the same! :>

#### Screenshot 1 -

```
from qiskit import QuantumCircuit, QuantumRegister, execute, Aer, transpile

circuit = QuantumCircuit(3)
circuit.iso([[0,1],[1,0]],[1],[]) # works fine 
GATEB2=transpile(circuit, basis_gates=['u1', 'u2', 'u3', 'cx', 'id']).to_gate() # works fine
CGA=GATEB2.control(1) # works fine too!

test_circuit = QuantumCircuit(4)
test_circuit.append(CGA,[0,1,2,3])
test_circuit.draw("mpl")
```
![Screenshot_87](https://user-images.githubusercontent.com/61380731/145852809-45731455-2d00-4a29-9451-0e6dc6597426.png)

Will update comment/add more comments if cause is found/fixed!

> Python version 3.9.7
> qiskit-terra version: 0.19.1

**Comment by jakelishman** [comment]
Funnily enough, this one is essentially the same issue as #7399.

**Comment by 1ucian0** [comment]
Nice! I just confirmed that #7403 fixes this issue.

**Comment by jakelishman** [comment]
This was fixed by a combination of #7146 and #10607.
